http/api/transport.go

```diff
@@ -27,35 +29,53 @@ import (
 const protocol = "http"
 
 var (
-	errMalformedData = errors.New("malformed request data")
-	auth             mainflux.ThingsServiceClient
+	errMalformedData  = errors.New("malformed request data")
+	auth              mainflux.ThingsServiceClient
+	channelPartRegExp = regexp.MustCompile(`^/channels/([\w\-]+)/messages((/[^/]+)*)*(\?.*)?$`)
```

Comment:
actually http does not use query parameters, i use the same regexp for coap/ws/http. If you do not want a specific regexp per adapter, it could be placed in a shared place

Reply:
@beres If you want to do this, `topics.go` file in the project root is a good place for that.

Reply:
@dusanb94 could also be the place for common subtopic function?

Reply:
@beres If that function is shared and does not accept any adapter-specific arguments, then I'd say yes.